src/lib.rs

```diff
@@ -74,49 +90,242 @@
+//! passing `RUSTFLAGS="--cfg loom"` as part of the command when you want to run the loom tests.
+//! Then modify your `Cargo.toml` to include loom like this:
+//!
+//! ```toml
+//! [target.'cfg(loom)'.dependencies]
+//! loom = "0.4"
+//! ```
+//!
+//! One common strategy to use the right types with and without loom is to create a module in your
+//! crate named `sync` or any other name of your choosing. In this module, list out the types that
+//! need to be toggled between loom and `std`:
+//!
+//! ```
+//! #[cfg(loom)]
+//! pub(crate) use loom::sync::atomic::AtomicUsize;
+//!
+//! #[cfg(not(loom))]
+//! pub(crate) use std::sync::atomic::AtomicUsize;
+//! ```
+//!
+//! Then, elsewhere in the library:
+//!
+//! ```
+//! use crate::sync::AtomicUsize;
+//! ```
+//!
+//! ## Handling Loom API differences.
+//!
+//! Most of loom's type are drop-in replacements for their counterpart in `std`, but sometimes
+//! there are minor API differences that you must work around. If your library must use Loom APIs
+//! that differ from `std` types, then the library will be required to implement those APIs for
+//! `std`. For example, for `UnsafeCell`, in the library's source, add the following:
+//!
+//! ```
+//! #![cfg(not(loom))]
 //!
-//! Tests must use the loom synchronization types, such as `Atomic*`, `Mutex`,
-//! `RwLock`, `Condvar`, `thread::spawn`, etc. When writing a concurrent program,
-//! the `std` types should be used when running the program and the `loom` types
-//! when running the test.
+//! #[derive(Debug)]
+//! pub(crate) struct UnsafeCell<T>(std::cell::UnsafeCell<T>);
 //!
-//! One way to do this is via cfg flags.
+//! impl<T> UnsafeCell<T> {
+//!     pub(crate) fn new(data: T) -> UnsafeCell<T> {
+//!         UnsafeCell(std::cell::UnsafeCell::new(data))
+//!     }
 //!
-//! It is important to not include other sources of non-determism in tests, such
-//! as random number generators or system calls.
+//!     pub(crate) fn with<R>(&self, f: impl FnOnce(*const T) -> R) -> R {
+//!         f(self.0.get())
+//!     }
 //!
-//! # Yielding
+//!     pub(crate) fn with_mut<R>(&self, f: impl FnOnce(*mut T) -> R) -> R {
+//!         f(self.0.get())
+//!     }
+//! }
+//! ```
 //!
-//! Some concurrent algorithms assume a fair scheduler. For example, a spin lock
-//! assumes that, at some point, another thread will make enough progress for
-//! the lock to become available.
+//! ## Yielding
 //!
-//! This presents a challenge for loom as the scheduler is not fair. In such
-//! cases, loops must include calls to `yield_now`. This tells loom that another
-//! thread needs to be scheduled in order for the current one to make progress.
+//! Some concurrent algorithms assume a fair scheduler. For example, a spin lock assumes that, at
+//! some point, another thread will make enough progress for the lock to become available. This
+//! presents a challenge for loom as the scheduler is not fair. In such cases, loops must include
```

Comment:
You could expand that, technically, the spinning thread doesn't ever have to yield back to the scheduler so loom will explore that option and blow the stack. This is also a good reason to not use spin locks :)

Reply:
And also to link to matklad's excellent https://matklad.github.io/2020/01/02/spinlocks-considered-harmful.html